File: Lab2/src/Domain/Matrix.java
package Domain;

import static Domain.AppUtils.randomNumber;

public class Matrix {

    private Integer lines = 0;
    private Integer columns = 0;
    private Integer[][] matrix;

    public Matrix(Integer lines, Integer columns, Boolean generate) {
        this.lines = lines;
        this.columns = columns;
        this.matrix = new Integer[lines][columns];
        if (generate)
            this.generateMatrix();
        else this.generateNullMatrix();
    }

    public Integer getNumberLines() {
        return lines;
    }

    public void setNumberLines(Integer lines) {
        this.lines = lines;
    }

    public Integer getNumberColumns() {
        return columns;
    }

    public void setNumberColumns(Integer columns) {
        this.columns = columns;
    }

    private void generateMatrix() {
        for (Integer l = 0; l < this.lines; l++)
            for (Integer c = 0; c < this.columns; c++)
                this.matrix[l][c] = randomNumber();
    }

    private void generateNullMatrix() {
        for (Integer l = 0; l < this.lines; l++)
            for (Integer c = 0; c < this.columns; c++)
                this.matrix[l][c] = 0;
    }


    public void printMatrix() {
        for (Integer l = 0; l < this.lines; l++) {
            for (Integer c = 0; c < this.columns; c++)
                System.out.print(this.matrix[l][c] + " ");
            System.out.println();
        }
    }

    public Integer getElement(Integer line, Integer column){
        return matrix[line][column];
    }

    public void setElement(Integer line, Integer column, Integer element){
        matrix[line][column] = element;
    }
}

File: Lab3_ThreadPoolC++/Lab3/Lab3/Matrix.cpp
#include <iostream>
#include "Matrix.h";
	static int randomNumber() {
		int lowBound = 1;
		int highBound = 5;
		return int(rand() % (highBound - lowBound + 1) + lowBound);
	}

	int Matrix::getElement(int line, int column) {
		return matrix[line][column];
	}

	void Matrix::setElement(int line, int column, int element) {
		matrix[line][column] = element;
	}

	int Matrix::getNumberLines() {
		return lines;
	}

	int Matrix::getNumberColumns() {
		return columns;
	}

	Matrix::Matrix(){}

	Matrix::Matrix(int l, int c, bool generate) {
		lines = l;
		columns = c;
		if (generate == true)
			generateMatrix();
		else generateNullMatrix();
	}

	void Matrix::generateMatrix() {
		for (int l = 0; l < getNumberLines(); l++)
			for (int c = 0; c < getNumberColumns(); c++)
				setElement(l, c, randomNumber());
	}

	void Matrix::generateNullMatrix() {
		for (int l = 0; l < getNumberLines(); l++)
			for (int c = 0; c < getNumberColumns(); c++)
				setElement(l, c, 0);
	}